viewing_party/main.py

```diff
@@ -0,0 +1,129 @@
+
+#the functions below are for wave_01
+# 
+
+def create_movie(movie_title,  genre, rating):
+    if not movie_title or not genre or not rating:
+        return None
+    return {
+        "title":  movie_title, 
+        "genre" : genre,
+        "rating" :rating 
+        }
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+  
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie):
+    watched_movie = {}
+    watchlist = user_data["watchlist"]
+    for index in range(len(watchlist)):
+        if watchlist[index]["title"] == movie:
+            watched_movie = watchlist[index]
+            watchlist.pop(index)
+            return add_to_watched(user_data, watched_movie)
+    return user_data
+
+#the functions below are for wave_02 
+def get_watched_avg_rating (user_data):
+    sum = 0 
+    watched = user_data["watched"]
+    for index in range(len(watched)):
+       sum += watched[index]["rating"]
+    if sum: return sum / len(watched)
+    return 0
+
+
+def get_most_watched_genre(user_data):
+    genres = {}
+    most_watched = {}
+    watched = user_data["watched"]
+    if len(watched)== 0:
+        return None
+    for index in range(len(watched)):
+        genre = watched[index]["genre"]
+        if genres.get(genre):
+            genres[genre] +=1 
+        else: genres[genre] = 1
+    key = list(genres.keys())[0]
+    value = list(genres.values())[0]
+    most_watched[key] = value
```

Comment:
quite a few variables! maybe we can pare it down maybe, like `if genres.get(watched[index]["genre"])`